dissect/target/plugins/apps/texteditor/windowsnotepad.py

```diff
@@ -0,0 +1,184 @@
+import zlib
+from typing import BinaryIO, Iterator
+
+from dissect import cstruct
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.plugin import export
+from dissect.target.plugins.apps.texteditor.texteditor import (
+    GENERIC_TAB_CONTENTS_RECORD_FIELDS,
+    TexteditorPlugin,
+)
+
+c_def = """
+struct multi_block_entry {
+    uint16    offset;
+    uleb128   len;
+    wchar     data[len];
+    char      crc32[4];
+};
+
+struct single_block_entry {
+    uint16    offset;
+    uleb128   len;
+    wchar     data[len];
+    char      unk1;
+    char      crc32[4];
+};
+
+struct header_crc {
+    char      unk[4];
+    char      crc32[4];
+};
+
+struct tab {
+    char                        magic[3];         // NP\x00
+    char                        header_start[2];  // \x00\x01
+    uleb128                     fsize1;
+    uleb128                     fsize2;
+    char                        header_end[2];    // \x01\x00
+
+    // Data can be stored in two says:
+    //  1. A single, contiguous block of data that holds all the data
+    //     In this case, the header is included in the single CRC32 checksum present at the end of the block
+    //  2. Multiple blocks of data that, when combined, hold all the data
+    //     In this case, the header has a separate CRC32 value stored at the end of the header
+    // The following bitmask operations basically check whether fsize1 is nonzero (boolean check) and depending
+    // on the outcome, parse 0 or 1 (so basically, parse or not parse) structs.
+    header_crc                  header_crc[((fsize1 | -fsize1) >> 31) ^ 1]; // Optional, only if fsize1 == 0
+    single_block_entry          single_block_entry[((fsize1 | (~fsize1 + 1)) >> 31) & 1];  // Optional, only if fsize1 > 0  # noqa: E501
+
+
+    multi_block_entry           multi_block_entries[EOF];  // Optional. If a single_block_entry is present
+                                                           // this will already be at EOF, so it won't do anything.
+                                                           // Otherwise, it will parse the individual blocks.
+};
+"""
+TextEditorTabRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "texteditor/windowsnotepad/tab", GENERIC_TAB_CONTENTS_RECORD_FIELDS
+)
+
+c_windowstab = cstruct.cstruct()
+c_windowstab.load(c_def)
```

Comment:
True, but all the data was stored in `AppData/Local/Packages/Microsoft.WindowsNotepad_` which means that it's the newer Windows Notepad App. It's different from the OG `notepad.exe` so that's why I made this distinction. Therefore I'd opt for just leaving it like this.

Reply:
The rest of the suggestion still applies 😄

Reply:
Ah right, now I see! :smile:

Reply:
Changed it now :)